[RA2 Ch03] 3.2.2.1 Network semantics (#2546)
cntt-n/CNTT: doc/ref_model/chapters/chapter03.md

* ITU-T Rec. G.8264 that describes the specification of Ethernet Synchronization Messaging Channel (ESMC)
SyncE architecture minimally requires replacement of the internal clock of the Ethernet card by a phase locked loop in order to feed the Ethernet PHY.

<a name="3.5.7"></a>

>  ### 3.5.7 Kubernetes Networking Semantics

Reviewer: This section is a duplicate of Section 3.2.2.1 above.

* ITU-T Rec. G.8264 that describes the specification of Ethernet Synchronization Messaging Channel (ESMC)
SyncE architecture minimally requires replacement of the internal clock of the Ethernet card by a phase locked loop in order to feed the Ethernet PHY.

<a name="3.5.7"></a>
 ### 3.5.7 Kubernetes Networking Semantics

> The support for traditional network orchestration is non existing in Kubernetes proper. Kubernetes is foremost a Platform as a Service (PaaS) environment and not an Infrastructure as a Service (Iaas) infrastructure component. There is no orchestration API like Neutron Openstack and no way to create L2 networks, instantiate network services such as L3aaS and LBaaS and then connect it all together like you can do with Neutron.

Reviewer 2: Some rewording
```suggestion
The support for traditional network orchestration is non existing in Kubernetes proper. Kubernetes is foremost a Platform as a Service (PaaS) environment and not an Infrastructure as a Service (Iaas) infrastructure component. There is no orchestration API like Neutron in Openstack and there is no way to create L2 networks, instantiate network services such as L3aaS and LBaaS and then a way connect them all together like in case of Neutron.
```

The basic semantics of Kubernetes and the information found in manifest defines the connectivity rules and behavior without in principle any references to IP addresses. This has many advantages, it makes it easy to create portable, scalable SW services and network policies for them that are not location aware and therefore can be executed more or less anywhere.

Kubernetes built in objects
Pod and workloads | Description
------------------|------------
[Pod:](https://kubernetes.io/docs/concepts/workloads/pods/) | Pod is a collection of containers that can run on a host. This resource is created by clients and scheduled onto hosts.
[ReplicaSet:](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/) | ReplicaSet ensures that a specified number of pod replicas are running at any given time.
[Deployment:](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) | Deployment enables declarative updates for Pods and ReplicaSets.
[DaemonSet:](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) | A Daemon set ensures that the correct nodes run a copy of a Pod.

> [Job:](https://kubernetes.io/docs/concepts/workloads/controllers/job/) | A Job represent a task, it will creates one more Pods and will continue to retry until the expected number of successful completions is reached.

Reviewer 2: ```suggestion
[Job:](https://kubernetes.io/docs/concepts/workloads/controllers/job/) | A Job represent a task, it creates one or more Pods and will continue to retry until the expected number of successful completions is reached.
```

spec:
        selector:
                matchLabels:
                        app: my-app
                        replicas: 4
                        template:
                                metadata:
                                        labels:
                                                app: my-app
                        spec:
                                containers:
                                        - name: my-app
                                          image: my-app-1.2.3
                                          ports:        
                                          - containerPort: 123 
```

This is all that is needed to deploy 4 pods/containers that are fronted by a service that performes load balancing. The *Deployment* will ensure that there are always four pods of type *my-app* available. the *Deployment* is responsible for the full lifecycle management of the pods, this includes in service update/upgrade.

None of this is of much help however when implementing network service functions such as VNFs/CNFs that requires multi networking and network orchestration

<a name="3.5.7.2"></a>
#### 3.5.7.2 Multi networking and Orchestration

> Kubernetes does currently not in itself support multi networks, pod multi network attachments or network orchestration. This is supported by using a chain loading [*Container Network Interface*](https://github.com/containernetworking/cni) plugin such as [Multus:](https://github.com/k8snetworkplumbingwg/multus-cni).

Reviewer 2: ```suggestion
Kubernetes does currently not in itself support multi networks, pod multi network attachments or network orchestration. This is supported by using a [*Container Network Interface*](https://github.com/containernetworking/cni) multiplexer such as [Multus](https://github.com/k8snetworkplumbingwg/multus-cni).
```
